src/Admin/assets/src/preview-status.ts

```diff
@@ -184,17 +192,45 @@ function onIframeLoadedHideLoaderUI(): void {
 	}, 50)
 }
 
-export async function doubleCheckIfPreviewFrontendIsOnline() {
-	const fetchResponse = await fetch(initialState.previewFrontendUrl)
+const requestPreviewFrontendIsOnline = async (): Promise<boolean> =>
+	(
+		await (
+			await fetch(`/?${initialState.graphqlEndpoint}`, {
+				method: "POST",
+				body: JSON.stringify({
+					query: previewFrontendIsOnlineQuery,
+				}),
+				headers: {
+					"Content-Type": "application/json",
+				},
+			})
+		).json()
+	)?.data?.wpGatsby?.isPreviewFrontendOnline
 
-	if (fetchResponse.ok) {
-		// if the response came back ok and we haven't already started loading the UI
-		if (!initialState.previewWebhookIsOnline) {
-			// start loading it because the frontend actually is online
-			await fetchPreviewStatusAndUpdateUI()
-		}
+let frontendOnlineCheckCount = 0
+
+/**
+ * If our backend webhook preview POST came back with an error, we can't be sure our frontend is online.
```

Comment:
I wonder if this comment should be updated. It seems like based on the code an error will be shown after 10 seconds + the request time of checking for the status 10 times.

Reply:
Good point! Just updated it. Thanks :D